components/mapLegend.js

```diff
@@ -1,51 +1,45 @@
-import React, { Component, useEffect, useMapEvents, useState } from "react";
-import styles from '../styles/legend.module.css'
-import AnimateHeight from 'react-animate-height';
-import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
-import { useSpring, animated } from "react-spring";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import React, { useState } from "react";
+import AnimateHeight from "react-animate-height";
+import { animated, useSpring } from "react-spring";
+import styles from "../styles/legend.module.css";
 
 export default function MapLegend({ location }) {
-    const [height, setHeight] = useState(0);
-    const [open, setOpen] = useState(false);
+  const [height, setHeight] = useState(0);
+  const [isOpen, setIsOpen] = useState(false);
 
-    const clickHandler = (e) => {
-        setOpen(!open)
-        setHeight(height === 0 ? 'auto' : 0)
-    }
+  const clickHandler = (e) => {
+    setIsOpen(!isOpen);
+    setHeight(height === 0 ? "auto" : 0);
```

Comment:
height type weird to me: is it a string or a number? maybe we should switch to typescript now xD

Reply:
Yes, switching to typescript is on the roadmap.